microsoft/botframework-solutions
# [Virtual Assistant Template][TypeScript] Add Virtual Assistant Template

## Description

- Add `Virtual Assistant Template`
   - Add **main functionality** for the `generator assistant`
   - Add `package.json`
   - Implement **TSLint**
   - Add basic configuration for the template
   - Add **deployment resources**
   - Add **new cards**
   - Add **dialogs** and **responses**
   - Add **locales**
   - Add **services folder**
   - Add **startup files** and **adapter**
   - Add `cognitiveModels` and `skills manifest`
   - Add `npm registry`
- Add **main functionality** for the `generator assistant`
- Add **tests** for the `generator assistant`

Related #1058

We made a parity between **C#** and **TypeScript** up to [this commit](https://github.com/Microsoft/AI/commit/74849d7bf9d5615fcf46ed9c59414527bfffa936) in [4.4 branch](https://github.com/Microsoft/AI/tree/4.4).

## Testing Steps
1. Go to `AI\templates\Virtual-Assistant-Template\src\typescript\generator-botbuilder-assistant`
2. Run `npm link` to symlink the package folder
3. Open a terminal wherever you want to create an assistant
4. Run `yo botbuilder-assistant`
5. Follow the generator's prompts
6. Check that the generator finished successfully
7. Run the `deploy.ps1` (which is inside `deployment\scripts` folder) being in the generated assistant folder
8. Verify that the deployment finished successfully
9. Run your generated assistant
10. Check that the assistant has the same behavior as C# version
	
## Checklist
